src/user_input_compiler.rs

```diff
@@ -245,13 +241,27 @@ fn collect_real(s: &mut Chars) -> Result<String, Input> {
   Ok( i1 )
 }
 
+fn match_full_integer(s :&mut Chars) -> Result<i64, Input> {
+  let _ = consume_whitespaces(s);
+  let sign = if let Some(c) = s.clone().next() { if c == '-' { -1 } else { 1 } } else { 1 };
```

Comment:
I'd change this to a multi-line expression or move it out into a function or macro. This one-liner is really hard to read :)